test/e2e/suite_test.go

```diff
@@ -93,11 +79,30 @@ var _ = BeforeSuite(func() {
 	k8sClient, err = client.New(cfg, client.Options{Scheme: scheme.Scheme})
 	Expect(err).NotTo(HaveOccurred())
 	Expect(k8sClient).NotTo(BeNil())
+
+	LwsReadyForTesting(k8sClient)
 })
 
 var _ = AfterSuite(func() {
 	cancel()
-	By("tearing down the test environment")
-	err := testEnv.Stop()
-	Expect(err).NotTo(HaveOccurred())
 })
+
+func LwsReadyForTesting(client client.Client) {
+	By("waiting for ")
+
+	// To verify that webhooks are ready, let's create a simple lws.
+	lws := testutils.BuildLeaderWorkerSet(metav1.NamespaceDefault).Replica(3).Obj()
+
+	// Once the creation succeeds, that means the webhooks are ready
+	// and we can begin testing.
+	Eventually(func() error {
+		return client.Create(ctx, lws)
+	}, timeout, interval).Should(Succeed())
+
+	// Delete this leaderworkerset before beginning tests.
+	Expect(client.Delete(ctx, lws))
```

Comment:
why do we delete the lws here before get?

Reply:
This lws serves no other purpose other than to ensure the webhooks are ready, once they are the lws is no longer needed here